### PR TITLE
Add ability to autofill a column with an sqlite function at record insertion time

### PR DESCRIPTION
**Pull request by adizaimi**
Add ability to autofill a column with an sqlite function at record insertion time. To have a field autoasigned set the dbstore parameter of a given field to the function desired. For instance to have a field autoassigned guid(), you would set `dbstore={guid()}`. Ddl syntax would be `CREATE TABLE T1("autoid" DEFAULT(guid()) PRIMARY KEY, i int)`.

The csc2 syntax is as follows:
```
    schema 
    {
        byte autoid[16] dbstore = {GUID()} null = no
        byte bytfld1[16] dbstore = {randomblob(16)} null = yes
        byte bytfld2[4] dbstore = x'00112233' null = yes
        cstring strfld[5] dbstore = {hex('hi')} null = yes
        int  i
    }
    keys
    {
            \"PK\"=autoid
    }
```
Example with DDL with insert/select:
```
cdb2sql adidb  'create table "t4"("autoid" byte(16) DEFAULT ( guid() ), "bt" byte(16) DEFAULT (randomblob(16)), "dt" datetime DEFAULT ( current_timestamp ), "rndmstr" char(19) DEFAULT ( hex(random()%10000) ), "i" int);'
cdb2sql adidb  'insert into t4(i) values(1),(2)'
(rows inserted=2)
cdb2sql adidb  'select * from t4'
(autoid=x'4f0b7c770c884543ad68d97a856b064d', bt=x'a3eb4668201314a0dee6d4674141f966', dt="2020-08-09T230539.105 America/New_York", rndmstr='2D35323132', i=1)
(autoid=x'a4a21d4155d541f5ba9dc7ecf9ecf225', bt=x'2cb4832ad130a7a21998b38d79d8c053', dt="2020-08-09T230539.105 America/New_York", rndmstr='31383330', i=2)
```

Please see the test in this checkin for more examples.

### CODE REVIEW

File: csc2/macc_so.c [hunk 1]
```diff
@@ -3125,10 +3140,18 @@ static int dyns_get_field_option_comn(char *tag, int fidx, int option,
             case CLIENT_BYTEARRAY: {
                 int *bytes;
                 int length;
+                if (*value_type == CLIENT_BYTEARRAY && vbsz >= tables[tidx].sym[fidx].szof && 
+                   tables[tidx].sym[fidx].szof == 16 && 
+                   strcasecmp(tables[tidx].sym[fidx].fopts[i].value.strval, "GUID()") == 0) {
```
Comment (nirbhayc) [inline csc2/macc_so.c:3145]: May be a comment that explains why we look for "GUID()" here and for "(GUID())" (in parens) above.

File: csc2/macclex.l [hunk 2]
```diff
@@ -226,7 +227,10 @@ sqlhexstr x{space}*"'"{space}*([a-fA-F0-9])*{space}*"'"
 
 
 <RECTYPE>dbpad                  { return T_FLD_PADDING; }
-<RECTYPE>dbstore                { return T_FLD_STRDEFAULT;}
+<RECTYPE>dbstore                { BEGIN DBSTORE; return T_FLD_STRDEFAULT;}
+<DBSTORE>[={()]			{ return yytext[0]; }
+<DBSTORE>"}"			{ BEGIN RECTYPE; return yytext[0]; }
```
Comment (nirbhayc) [inline csc2/macclex.l:232]: misaligned?

File: docs/pages/programming/table_schema.md [hunk 3]
```diff
@@ -93,6 +94,7 @@ The dbstore value must have the same datatype as the column they are attached to
 for current database system timestamp.
 * For longlong integer fields may specify a dbstore of nextsequence.  This will populate the column with a value one greater than the largest value which has ever been seen for the column.
 * For byte arrays you can specify dbstore=0 to indicate that it should be zeroed be default.
+* If the type is byte[16], you can also specify `dbstore="GUID"` and that will allow the field to be autofilled by the database at the time of recond insertion.
```
Comment (nirbhayc) [inline docs/pages/programming/table_schema.md:97]: dbstore={GUID()}

[the remark this answers is not in the document]
Comment (adizaimi) [reply docs/pages/programming/table_schema.md:97]: fixing

Comment (adizaimi) [reply docs/pages/programming/table_schema.md:97]: fixing

File: sqlite/src/comdb2build.c [hunk 4]
```diff
@@ -3153,10 +3153,16 @@ static char *format_csc2(struct comdb2_ddl_context *ctx)
...
             */
-            if ((type_flags[column->type] & FLAG_QUOTE_DEFAULT) != 0) {
-                strbuf_appendf(csc2, "dbstore = \"%s\" ", column->def);
-            } else {
+            if (*column->def == '(') {
+                /* turn DEFAULT(funct()) into dbstore={func()} */
+                int len = strlen(column->def);
+                assert(column->def[len - 1] == ')');
+                strbuf_appendf(csc2, "dbstore = {%.*s} ", len - 2, column->def + 1);
+            } else if (*column->def == '(' ||
```
[the remark this answers is not in the document]
Comment (adizaimi) [reply sqlite/src/comdb2build.c:3161]: yes, putting as it was.

File: csc2/macclex.l (hunk 2 above)
Comment (adizaimi) [reply csc2/macclex.l]: fixed

File: csc2/macc_so.c (hunk 1 above)
Comment (adizaimi) [reply csc2/macc_so.c]: this was deleted.

**Pull request merged**